### PR TITLE
shader/execution: Add more constructor tests

### PR DESCRIPTION
**Pull request by ben-clayton**
Add abstract non-zero constructor tests.
Add structure zero and non-zero constructor tests.

Issue: #1606
Issue: #1705

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/constructor/non_zero.spec.ts [hunk 1]
```diff
@@ -141,9 +161,39 @@ g.test('vector_splat')
...
+    }
+  })
+  .fn(async t => {
+    const suffix = t.params.abstract_type === 'abstract-float' ? '.0' : '';
+    const concreteElementType = Type[t.params.concrete_type];
+    const concreteVectorType = Type.vec(t.params.width, concreteElementType);
+    const fn = `vec${t.params.width}`;
+    await run(
+      t,
+      basicExpressionBuilder(_ => `${fn}(${t.params.value * 0x100000000}${suffix}) / 0x100000000`),
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/webgpu/shader/execution/expression/constructor/non_zero.spec.ts:186]: Correct.

File: src/webgpu/shader/execution/expression/constructor/non_zero.spec.ts [hunk 2]
```diff
@@ -184,25 +234,63 @@ g.test('vector_elements')
...
 
-g.test('vector_mix')
+g.test('abstract_vector_elements')
+  .specURL('https://www.w3.org/TR/WGSL/#value-constructor-builtin-function')
+  .desc(`Test that a vector constructed from abstract element values produces the expected value`)
+  .params(u =>
+    u
+      .combine('abstract_type', ['abstract-int', 'abstract-float'] as const)
+      .expand('concrete_type', t => kConcreteTypesForAbstractType[t.abstract_type])
+      .combine('value', [1, 2, 5, 100] as const)
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/webgpu/shader/execution/expression/constructor/non_zero.spec.ts]: Good catch. Done.

File: src/webgpu/shader/execution/expression/constructor/non_zero.spec.ts [hunk 3]
```diff
@@ -380,9 +567,54 @@ g.test('matrix_column_vectors')
...
+g.test('abstract_matrix_column_vectors')
   .specURL('https://www.w3.org/TR/WGSL/#value-constructor-builtin-function')
-  .desc(`Test that an array constructed from element values produces the expected value`)
+  .desc(`Test that a matrix constructed from abstract column vectors produces the expected value`)
+  .params(u =>
+    u
+      .combine('concrete_type', ['f32', 'f16'] as const)
+      .combine('columns', [2, 3, 4] as const)
+      .combine('rows', [2, 3, 4] as const)
+      .combine('infer_type', [false, true] as const)
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/webgpu/shader/execution/expression/constructor/non_zero.spec.ts]: Nope. Again, good catch. Removed.

**Review by dneto0 — APPROVED** [review]
Thanks!

**Pull request merged**